keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -0,0 +1,920 @@
+(ideally, under SIG Auth) in order to make it the canonical API for managing references
+across namespaces.
+
+### Goals
+
+* Move ReferenceGrant to an API Group that SIG Auth manages
+* Clearly define how ReferenceGrant should be used, including both current use
+  cases and guidance for future use cases
+* Implement a library to ensure that ReferenceGrant is implemented consistently
+  by all controllers
+
+### Non-Goals
+
+* Develop an authorizer that will automatically implement ReferenceGrant for all
+  use cases. (It would be impossible to represent concepts like "all namespaces"
+  or label selectors that have become important for this KEP).
+
+## Proposal
+
+Move the existing ReferenceGrant resource into a new
+`authorization.k8s.io` API group, defined within the Kubernetes code base
+as part of the 1.27 release.
+
+We will take this opportunity to clarify and update the API after SIG Auth
+feedback. This resource will start with v1alpha1 as the API version.
+
+
+### Risks and Mitigations
+
+#### No Default Implementation
+Similar to the Ingress and Gateway APIs, this API will be dependent on
+implementations by controllers that are not included by default in Kubernetes.
+This could lead to confusion for users. We'll need to rely heavily on
+documentation for this feature, tracking all uses of official Kubernetes APIs
+that support ReferenceGrant in a central place.
+
+#### Potential for Variations Among Implementations
+Because this relies on each individual controller to implement the logic,
+it is possible that implementations may become inconsistent. To avoid that,
+we'll provide a standard library for implementing ReferenceGrant. We'll
+also strongly recommend that every API that relies on ReferenceGrant 
+includes robust conformance tests covering this functionality. Existing
+Gateway API conformance tests can serve as a model for this.
+
+#### Cross-Namespace References may Weaken Namespace Boundaries
+Although we believe that the handshake required for cross-namespace references
+with ReferenceGrant ensures these references will be safe, it does potentially
+weaken existing namespace boundaries. We believe ReferenceGrant will have a net
+benefit on the ecosystem as it will allow workloads, secrets, and configuration
+to be deployed in separate namespaces that more clearly match up with desired
+authorization.
+
+## Design Details
+
+### General Notes
+
+#### `ReferenceGrant` is half of a handshake
+
+When thinking about ReferenceGrant, it is important to remember that it does not
+do anything by itself. It *Grants* the *possibility* of making a *Reference*
+across namespaces. It's intended that _another object_ (that is, the From object)
+complete the handshake by creating a reference to the referent object (the To
+object).
+
+#### ReferenceGrant authors must have sufficient access
+
+Anyone creating or updating a ReferenceGrant MUST have read access to the resources
```

Comment:
add an unresolved block (linking to https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084528657 as context) about whether checks beyond read are needed (if not all grants are limited to granting read access)

Reply:
Updated, thanks!